### PR TITLE
add support for osm and osm-edge.

### PR DESCRIPTION
**Pull request by cybwan**
[Open Service Mesh (OSM)](https://github.com/openservicemesh/osm) is a lightweight, extensible, cloud native [service mesh](https://en.wikipedia.org/wiki/Service_mesh) that allows users to uniformly manage, secure, and get out-of-the-box observability features for highly dynamic microservice environments.

[Open Service Mesh Edge (OSM-Edge)](https://github.com/flomesh-io/osm-edge) fork from [Open Service Mesh](https://github.com/openservicemesh/osm), built purposely for Edge computing. [OSM edge](https://github.com/flomesh-io/osm-edge) uses lightweight, Cloud Native, programmable proxy [Pipy](https://flomesh.io/) as sidecar proxy.

PR is bundled with Unit Test (UT) and E2E test suite to cover the changes.

signed-off-by: cybwan <baili@flomesh.io>

### CODE REVIEW

**Comment by kebe7jun** [comment]
wow, awesome!

File: deploy/README.md [hunk 1]
```diff
@@ -52,6 +52,15 @@ helm install -n linkerd --set mode=linkerd merbridge helm
 helm install -n kuma-system --set mode=kuma merbridge helm
 ```
 
+### Install Merbridge on Osm/Osm-edge
+
++ set mode=osm to switch to osm/osm-edge mode
++ set namespace where merbridge is going to install by -n .
```
Comment (windsonsea) [inline deploy/README.md:58]:
```suggestion
### Install Merbridge on OSM/OSM-Edge

+ Set `mode=osm` to switch to [OSM](https://github.com/openservicemesh/osm)/[OSM-Edge](https://github.com/flomesh-io/osm-edge) mode
+ Specify a namespace where you want to install Merbridge using `-n`
```

**Review by windsonsea — COMMENTED** [review]
Not sure the link is correct or not

File: deploy/README.md (hunk 1 above)
Comment (cybwan) [reply deploy/README.md:58]: correct.

File: controller/pod.go [hunk 2]
```diff
@@ -283,6 +288,86 @@ func parsePodConfigFromAnnotationsKuma(annotations map[string]string, pod *podCo
...
+func parsePodConfigFromAnnotationsOsm(annotations map[string]string, pod *podConfig) {
+	statusPort := 15021
+	if v, ok := annotations["openservicemesh.io/port"]; ok {
+		vv, err := strconv.ParseUint(v, 10, 16)
+		if err == nil {
+			statusPort = int(vv)
+		}
+	}
+	pod.statusPort = uint16(statusPort)
+	excludeInboundPorts := []uint16{15000, 15001, 15003, 15010, 15021, 15050, 15128, 15901, 15902, 15903, 15904}
```
Comment (kebe7jun) [inline controller/pod.go:300]: Currently, `MaxItemLen` is 10, maybe can not store too many ports.

Increase the `MaxItemLen` if it necessary.

**Review by kebe7jun — COMMENTED** [review]
Awesome! Almost lgtm if e2e passed.

File: controller/pod.go [hunk 3]
```diff
@@ -84,7 +84,7 @@ func createLocalPodController(client kubernetes.Interface) pods.Watcher {
 	}
 }
 
-const MaxItemLen = 10 // todo changeme
+const MaxItemLen = 12 // todo changeme
```
Comment (kebe7jun) [inline controller/pod.go:87]: also update https://github.com/merbridge/merbridge/blob/25f37a9051ba3b400ae0a84b52875caea5945fd6/bpf/headers/helpers.h#L257

we can set it to 20~

**Comment by kebe7jun** [comment]
failed reason:

https://github.com/merbridge/merbridge/blob/25f37a9051ba3b400ae0a84b52875caea5945fd6/bpf/Makefile#L94

value size need be updated manually.

File: controller/pod.go (hunk 2 above)
Comment (dddddai) [reply controller/pod.go:300]: BTW I think 15021 is overlapped with `statusPort`, maybe we should merge them in the future

**Review by kebe7jun — COMMENTED** [review]
lgtm @dddddai @Xunzhuo ptal.

**Review by dddddai — APPROVED** [review]
Thank you!

**Review by kebe7jun — APPROVED** [review]
Thank you.

**Pull request merged**